core/src/main/resources/hudson/model/BuildTimelineWidget/build-timeline-widget.js

```diff
@@ -11,19 +11,29 @@ function getData(eventSource1, current, min, max) {
   }
   if (!eventSource1.loaded[current]) {
     eventSource1.loaded[current] = true;
-    new Ajax.Request("timeline/data/", {
-      method: "POST",
-      parameters: { min: current * interval, max: (current + 1) * interval },
-      onSuccess: function (t) {
-        if (t.status != 0) {
-          try {
-            eventSource1.loadJSON(JSON.parse(t.responseText), ".");
+    fetch(
+      "timeline/data/?" +
+        new URLSearchParams({
+          min: current * interval,
+          max: (current + 1) * interval,
+        }),
+      {
+        method: "POST",
+        headers: {
+          [document.head.dataset.crumbHeader]: document.head.dataset.crumbValue,
+        },
```

Comment:
This unnecessarily duplicates the names of the `crumb-header` and `crumb-value` data attributes defined in `hudson-behavior.js`. For a full justification of the reasoning here, consult the following page:

https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

By invoking the functionality in the existing crumb subsystem, we can avoid the need for this repetition:

```suggestion
        headers: crumb.wrap({}),
```